torch/nn/parallel/distributed.py

```diff
@@ -396,10 +396,13 @@ def __init__(
 
         self.is_multi_device_module = len({p.device for p in module.parameters()}) > 1
         distinct_device_types = {p.device.type for p in module.parameters()}
-        assert len(distinct_device_types) == 1, (
-            "DistributedDataParallel's input module must be on "
-            "the same type of devices, but input module parameters locate in {}."
-        ).format(distinct_device_types)
+        if len(distinct_device_types) != 1:
```

Comment:
There is an assert on line 389 as well and another one in `_passing_sync_batchnorm_handle`.